src/System.Management.Automation/engine/Modules/ModuleIntrinsics.cs

```diff
@@ -1261,9 +1261,24 @@ internal static string GetWindowsPowerShellModulePath()
             var modulePathList = new List<string>();
             foreach (var path in currentModulePath.Split(';'))
             {
-                if (!excludeModulePaths.Contains(path))
+                var trimmedPath = path.Trim();
+                if (!excludeModulePaths.Contains(trimmedPath))
                 {
-                    modulePathList.Add(path);
+                    // make sure this module path is Not part of other PS Core installation
+                    var possiblePwshDir = Path.GetDirectoryName(trimmedPath);
+
+                    if (string.IsNullOrEmpty(possiblePwshDir))
+                    {
+                        // i.e. module dir is in the drive root
+                        modulePathList.Add(trimmedPath);
+                    }
+                    else
+                    {
+                        if (!File.Exists(Path.Combine(possiblePwshDir, "pwsh.exe")))
```

Comment:
It may be a nitpick here, but what if a parent PowerShell session starts with the FxDependent package where only pwsh.dll is present?
How about checking for `pwsh.dll`?

Reply:
Think about the powershell global tool.

Reply:
Is there a need to consider things built with the SDK in general here too? Those will have a concept of the module path.

Reply:
Very good point. Updated to check for `pwsh.dll`.

Reply:
Good question. That seems to be right in terms of completeness, but I'm not sure if it's really necessary.
@SteveL-MSFT @anmenaga thoughts?

Reply:
Ideally it is a nice to have but kind of a corner case. How difficult is it to detect a module path of SDK-built app just by looking at the file system?

This bug manifests itself mostly with modules with names of PS engine modules and running correct/old WindowsPS commands.
What is the the probability of an app built with the SDK having a module (in it's own module folder) with name like `"Microsoft.PowerShell.Management` or `Microsoft.PowerShell.Utility` in addition to this app wanting to use WindowsPS to run some old commands? Kind of rare.

Reply:
@anmenaga All applications that built against `Microsoft.PowerShell.SDK` nuget package can potentially have modules `Microsoft.PowerShell.Management` and `Microsoft.PowerShell.Utility` in their `$pshome\modules` path. This is because we ship all built-in modules along with `Microsoft.PowerShell.SDK` since 7.0.
For example, PowerShell Jupyter kernel have all those built-in modules in the `$pshome\modules` folder.

With that being said, the real question is -- what is the chance to start `pwsh` from one of those applications (e.g. PS Jupyter) and then start powershell.exe from that `pwsh`? This feels unlikely to me. So I'm fine with just changing to `pwsh.dll`. @SteveL-MSFT can decide if we really want to go 100% completeness.

Reply:
Every time I see once again a change in this code, I understand that I will never understand :-) why to be based on the variable change if this is a fundamentally insoluble problem in an SxS scenarios. What if there are several SDK applications in the row? There are an infinite number of application, modules and combinations.

And more tricky question: if PowerShell engine does the variable manipulation _should all SDK applications follow the logic to get compatible behavior_?

I still think it is better for PowerShell Engine to import/consume the variable values but does not export it - this should be delegated to applications/users who better know how address their specific scenario. 
If we think about services, then everything becomes easier as there are independent processes. It is unclear why then complicate the behavior of subprocesses.

Reply:
@iSazonov I personally agree with you, but the consensus is that we must proceed for now in a way that offers maximum compatibility with the existing concepts in PowerShell versions being used in the wild (particularly PS 5.1).

I worry that trying to nip and tuck the PSModulePath variable around other processes in the process stack is always going to have edge cases, but for something so fundamental to PowerShell this seems to be the more conservative option for now.

As for SDK embedding, I wonder if a scenario like `Start-Job` is going to be motivating. I'm not sure of what exact assemblies are published with the SDK, but we've defined `$PSHome` as where `System.Management.Automation.dll` lives and my thought is that perhaps that's the right assembly to anchor to...

Reply:
>  I personally agree with you, but the consensus is

Thanks :-).  For me it is a breaking change in grace area: nobody pointed me any affected critical scenario. I am amazed that team does not even start offer alternatives. My proposal was to expose $PSModulePath variable and GetPSModulePath() public method. Then 7.1 is nice milestone to remove these tricky PSModulePath env manipulations or make it _opt-in_. Otherwise, this behavior will always be problematic for years to come.

Reply:
> Then 7.1 is nice milestone to remove these tricky PSModulePath env manipulations or make it opt-in. Otherwise, this behavior will always be problematic for years to come.

Very much agreed. I'm not sure we have this work on the board for the coming milestone, but I'll see if I can get the team to discuss it.